app/services/moves/exporter.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Moves
+  class Exporter
+    attr_reader :moves
+
+    HEADINGS = [
+      'Status',
+      'Reference',
+      'Move type',
+      'Created at',
+      'Last updated',
+      'From location name',
+      'From location code',
+      'To location name',
+      'To location code',
+      'Additional information',
+      'Date of travel',
+      'PNC number',
+      'Prison number',
+      'Last name',
+      'First name(s)',
+      'Date of birth',
+      'Gender',
+      'Ethnicity',
+      'Ethnicity code',
+      'Violent',
+      'Violent details',
+      'Escape',
+      'Escape details',
+      'Must be held separately',
+      'Must be held separately details',
+      'Self harm',
+      'Self harm details',
+      'Concealed items',
+      'Concealed items details',
+      'Any other risks',
+      'Any other risks details',
+      'Special diet or allergy',
+      'Special diet or allergy details',
+      'Health issue',
+      'Health issue details',
+      'Medication',
+      'Medication details',
+      'Wheelchair user',
+      'Wheelchair user details',
+      'Pregnant',
+      'Pregnant details',
+      'Any other requirements',
+      'Any other requirements details',
+      'Solicitor or other legal representation',
+      'Solicitor or other legal representation details',
+      'Sign or other language interpreter',
+      'Sign or other language interpreter details',
+      'Any other information',
+      'Any other information details',
+      'Not for release',
+      'Not for release details',
+      'Not to be released',
+      'Not to be released details',
+      'Requires special vehicle',
+      'Requires special vehicle details',
+      'Uploaded documents',
+    ].freeze
+
+    MOVE_INCLUDES = [:from_location, :to_location, :profile, :documents, person: %i[gender ethnicity]].freeze
+
+    def initialize(moves)
+      @moves = moves
+    end
+
+    def call
+      Tempfile.new('export', Rails.root.join('tmp')).tap do |file|
```

Comment:
I am intentionally not cleaning this file up explicitly - as we need to access it in the controller actions. Tempfiles do get deleted during garbage collection though so shouldn't be an issue. See https://ruby-doc.org/stdlib-2.7.1/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close